recipe/build.sh

```diff
@@ -15,22 +15,28 @@ for i in `ls`; do
         # Headers and libraries are installed to targetsDir
         mkdir -p ${PREFIX}/${targetsDir}
         mkdir -p ${PREFIX}/$i
-        cp -rv $i ${PREFIX}/${targetsDir}
         if [[ $i == "bin" ]]; then
+            mkdir -p ${PREFIX}/${targetsDir}/bin
+            mv -v $i/nvcc ${PREFIX}/${targetsDir}/bin
+            cp -rv $i ${PREFIX}
             # Use a custom nvcc.profile to handle the fact that nvcc is a symlink.
-            cp ${RECIPE_DIR}/nvcc.profile.for_prefix_bin ${PREFIX}/bin/nvcc.profile
+            # cp ${RECIPE_DIR}/nvcc.profile.for_prefix_bin ${PREFIX}/bin/nvcc.profile
```

Comment:
Do we still need some variation of this for CMake to work? It is causing issues in the current state.

Reply:
Guessing we might need to rewrite the paths as shown below. I made `TOP` point to `$PREFIX` so that `$PREFIX/nvvm` can be found properly, and adapted includes/libraries by adding `$(_TARGET_DIR_)/` accordingly. (I haven't tested this locally but will give it a shot.)

```
TOP              = $(_HERE_)/../

NVVMIR_LIBRARY_DIR = $(TOP)/$(_NVVM_BRANCH_)/libdevice

LD_LIBRARY_PATH += $(TOP)/$(_TARGET_DIR_)/lib:
PATH            += $(TOP)/$(_TARGET_DIR_)/bin:$(TOP)/$(_NVVM_BRANCH_)/bin:$(_HERE_):

INCLUDES        +=  "-I$(TOP)/$(_TARGET_DIR_)/include" $(_SPACE_)

LIBRARIES        =+ $(_SPACE_) "-L$(TOP)/$(_TARGET_DIR_)/lib/stubs" "-L$(TOP)/$(_TARGET_DIR_)/lib"

CUDAFE_FLAGS    +=
PTXAS_FLAGS     +=
```

Reply:
Guessing we might need to rewrite the paths as shown below. I made `TOP` point to `$PREFIX` so that `$PREFIX/nvvm` can be found properly, and adapted includes/libraries by adding `$(_TARGET_DIR_)/` accordingly. (I haven't tested this locally but will give it a shot.)

```
TOP              = $(_HERE_)/../

NVVMIR_LIBRARY_DIR = $(TOP)/$(_NVVM_BRANCH_)/libdevice

LD_LIBRARY_PATH += $(TOP)/$(_TARGET_DIR_)/lib:
PATH            += $(TOP)/$(_TARGET_DIR_)/bin:$(TOP)/$(_NVVM_BRANCH_)/bin:$(_HERE_):

INCLUDES        +=  "-I$(TOP)/$(_TARGET_DIR_)/include" $(_SPACE_)

LIBRARIES        =+ $(_SPACE_) "-L$(TOP)/$(_TARGET_DIR_)/lib/stubs" "-L$(TOP)/$(_TARGET_DIR_)/lib"

CUDAFE_FLAGS    +=
PTXAS_FLAGS     +=
```

Reply:
We need `TOP` to be `$(_HERE_)/../$(_TARGET_DIR_)/` The NVVM on the path should be captured by `$(TOP)/$(_NVVM_BRANCH_)/bin` in the PATH

Reply:
Looks to be working again with this custom `nvcc.profile`, just needed to add `$PREFIX/targets/{{ target_name }}/nvvm` as a symlink to `$PREFIX/nvvm`

Reply:
> just needed to add `$PREFIX/targets/{{ target_name }}/nvvm` as a symlink to `$PREFIX/nvvm`

I wondered about this. Should it go the other way? i.e. `$PREFIX/nvvm` linking to `$PREFIX/targets/{{ target_name }}/nvvm`?

Reply:
IMO `$PREFIX/targets/{{ target_name }}/nvvm` is a WAR of an `nvcc` WAR. So whenever we eventually split `nvvm` to its own component `nvvm` would only install to `$PREFIX/nvvm` and `$PREFIX/targets/{{ target_name }}/nvvm` could continue to be a symlink owned by `cuda-nvcc`.